### PR TITLE
Catalog Of My Things: An overview of what we did.

### PR DESCRIPTION
**Pull request by Kingjosh007**
Here is an overview of what [Timo](https://github.com/Timowest12), [King Josaphat](https://github.com/Kingjosh007), and [Oluwafemi](https://github.com/phelian23) did as partners in this Ruby capstone project:

> We had to meet [the following requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/catalog_of_my_things.md)

## Video

We recorded a demo video. It is available [here](https://drive.google.com/file/d/15nVoYKsxtMaHj99W5Z1l7_Egd49RJ5nR/view).

## Features 

As a user:
   * I can list my items (books, music albums, and games)
   * I can add new items to my catalog (books, music albums, and games)
   * I can check all the labels, genres, authors, and sources of my items
   * I can exit the program from the menu
   * I can see the data I previously saved every time I run the app
   * [BONUS] I have a nice UI with colors in the console.

## Workload distribution

 - [Timo](https://github.com/Timowest12) was responsible for everything related to **music albums** and **genres**;
 - [King Josaphat](https://github.com/Kingjosh007) was responsible for everything related to **books** and **labels**;
 - [Oluwafemi](https://github.com/phelian23) was responsible for everything related to **games** and **authors**;
 - Each of us had to write tests and some code in `schema.sql`;
 - Each of us had to give and receive code reviews to the other team members.

### CODE REVIEW

File: classes/game.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+                      end
+    @multiplayer = multiplayer
+  end
+
+  def can_be_archived?
+    nb_secs_in_year = 365.25 * 24 * 60 * 60
+    diff_in_years = ((Date.today.to_time - @last_played_at.to_time) / nb_secs_in_year)
+    super && diff_in_years > 2
+  end
+end
```
[the remark this answers is not in the document]
Comment (Kingjosh007) [reply classes/game.rb:23]: Fixed.

File: classes/label.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    @id = my_id
+    @title = title
+    @color = color
+  end
+
+  def add_item(item)
+    item.assign_label(self)
+    @@items << item unless @@items.include?(item)
+  end
+end
```
[the remark this answers is not in the document]
Comment (Kingjosh007) [reply classes/label.rb:19]: Solved. This class is not testing in `rspec/label_spec.rb`.

File: spec/source_spec.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    expect(@source.name).to eq('Internet')
+  end
+
+  it 'The add_items method should add item to @@items' do
+    item1 = Item.new(Date.today.to_s, true)
+    @source.add_item(item1)
+    items = Source.class_variable_get(:@@items)
+    expect(items).to include(item1)
+  end
+end
```
Comment (julie-ify) [inline spec/source_spec.rb:22]: - [ ]  Good job adding this test. However, this test is failing, let's try to fix it.

chrome_c3Z8U5C7of

File: spec/label_spec.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    it 'The add_items method should add item to @@items' do
+      label = Label.new('Whatever', 'red')
+      item1 = Item.new(Date.today.to_s, true)
+
+      label.add_item(item1)
+      items = Label.class_variable_get(:@@items)
+      expect(items).to include(item1)
+    end
+  end
+end
```
Comment (julie-ify) [inline spec/label_spec.rb:15]: - [ ] Good job adding this test. However, this test is failing, let's try to fix it.

chrome_lDZ1YGxG59

File: spec/genre_spec.rb [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    @name == 'country'
+  end
+
+  it 'The add_items method should add item to @@items' do
+    item1 = Item.new(Date.today.to_s, true)
+    @genre.add_item(item1)
+    items = Genre.class_variable_get(:@@items)
+    expect(items).to include(item1)
+  end
+end
```
Comment (julie-ify) [inline spec/genre_spec.rb:22]: - [ ]  Good job adding this test. However, this test is failing, let's try to fix it.

chrome_55BBV8cX36

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Kingjosh007, @Timowest12, and @phelian23,

Good job so far!

Hello, good job working on some of the requested changes so far 👍. However, there are still issues that haven't been addressed yet but you are almost there.

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by eri8-9 — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

- Well done!

Felix

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**